Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ cmake_dependent_option(RESOURCE_BRUTE "Allow pngcrush brute-force optimization"
 if(PLASMA_BUILD_RESOURCE_DAT)
     set(external_DAT "${CMAKE_CURRENT_BINARY_DIR}/resource.dat")
     if(RESOURCE_OPTIMIZE)
-        string(APPEND OPTIMIZE_ARGUMENT "--pngcrush \"${PNGCRUSH_EXECUTABLE}\" ")
+        string(APPEND OPTIMIZE_ARGUMENT "--pngcrush=\"${PNGCRUSH_EXECUTABLE}\"")
```

Comment:
We need to keep the space at the end, because it appends other arguments to the string without a leading space

Reply:
I believe that causes things to fail again - but lemme double check.

Reply:
Does actually seem to work on this machine even with the space. I'll make the edit.

Reply:
I take it back. The space is once again not working.

```
Showing All Messages
FileNotFoundError: [Errno 2] No such file or directory: '/Applications/Xcode.app/Contents/Developer/usr/bin/pngcrush '
```

See how the space is incorporated into the file path?

As I was researching this I found some information that CMake does not do simple string work very well and will try to automatically escape things or encapsulate them in quotes. That seems like this is what is happening here but I don't know enough to make a good guess as to what the best fix is.

Reply:
Can you print out the value of `PNGCRUSH_EXECUTABLE` from inside the CMake script to check if perhaps the space is already there in the variable and not caused by the argument building? In my experience, CMake *usually* handles string quoting/escaping very well - so if the argument passed to Python has a trailing space, that probably means that there's an actual space at the end of the variable value.

I wonder if it's a problem that `OPTIMIZE_ARGUMENT` is a single string of space-separated arguments, not a proper CMake list. Does the behavior change if you replace `string(APPEND ...)` with `list(APPEND ...)` and make sure there are no spaces *inside* the quoted strings? (For that to work, you *might* have to pass `COMMAND_EXPAND_LISTS` to [`add_custom_command`](https://cmake.org/cmake/help/latest/command/add_custom_command.html), but I'm not sure...)